### PR TITLE
Rover: Add support for MAV_CMD_DO_PAUSE_CONTINUE command in GUIDED and AUTO modes

### PR DESCRIPTION
**Pull request by mustafa-gokce**
Implemented MAV_CMD_DO_PAUSE_CONTINUE command handling in GUIDED and AUTO modes in Rover.
Can be tested using autotest.
To test pause/continue in AUTO mode: `./Tools/autotest/autotest.py build.Rover test.Rover.PauseContinue --map`
To test pause/continue in GUIDED mode: `./Tools/autotest/autotest.py build.Rover test.Rover.PauseContinueGuided --map`

### CODE REVIEW

File: Rover/mode_auto.cpp [hunk 1]
```diff
@@ -812,3 +812,34 @@ void ModeAuto::do_guided_limits(const AP_Mission::Mission_Command& cmd)
...
+    // stop the vehicle
+    if (rover.is_boat()) {
+        if (!start_loiter()) {
+            start_stop();
+        }
+    } else {
+        start_stop();
+    }
+
+    // stop the mission
```
Comment (peterbarker) [inline Rover/mode_auto.cpp:832]: pointless comment

File: Rover/mode_auto.cpp [hunk 2]
```diff
@@ -812,3 +812,34 @@ void ModeAuto::do_guided_limits(const AP_Mission::Mission_Command& cmd)
...
+    mission.stop();
+
+    return true;
+}
+
+// resume auto mode
+bool ModeAuto::resume()
+{
+    // resume the mission
```
Comment (peterbarker) [inline Rover/mode_auto.cpp:841]: pointless comment

**Review by peterbarker — CHANGES_REQUESTED** [review]
Still looks good.

Same comment that I mentioned in the Copter PR about using structures to record information for each type of destination applies.

**Comment by tridge** [comment]
@rmackay9 needs approval from you

**Comment by mikerob** [comment]
Hi, interested in this functionality but have a couple of questions:
1) When you pause, does the vehicle continue following the waypoint path until it stops?  Does it gracefully handle passing waypoints as it slows?
2) When it resumes, does it return to the WP_SPEED or the last DO_CHANGE_SPEED?

**Comment by mustafa-gokce** [comment]
@mikerob 

> 1. When you pause, does the vehicle continue following the waypoint path until it stops?  Does it gracefully handle passing waypoints as it slows?

No. It is just an interim solution for pausing. It should be implemented with SCurves later for gracefully stopping along the path.



> 2\. When it resumes, does it return to the WP_SPEED or the last DO_CHANGE_SPEED?

I need to test this but probably resumes with the last DO_CHANGE_SPEED since the above logic doesn't enforce, save or change anything related to speed.

File: Rover/mode_guided.cpp [hunk 3]
```diff
@@ -235,6 +253,12 @@ bool ModeGuided::get_desired_location(Location& destination) const
 bool ModeGuided::set_desired_location(const struct Location& destination,
                                       float next_leg_bearing_cd)
 {
+    // set previous command
+    prev_cmd.prev_cmd_type = PrevCmd::SET_DESIRED_LOCATION;
+    prev_cmd.u.set_desired_location.destination = destination;
+    prev_cmd.u.set_desired_location.next_leg_bearing_cd = next_leg_bearing_cd;
```
Comment (mustafa-gokce) [inline Rover/mode_guided.cpp:259]: I forgot to save this, so updated the PR.

**Pull request closed without merging**